File: 1/2020.09.22/429.N-ary Tree Level Order Traversal/java/solution.java
class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> res = new ArrayList<>();
        levelOrder(root, res, 0);
        return res;
    }

    private void levelOrder(Node x, List<List<Integer>> res, int height) {
        if (x == null) {
            return;
        }
        if (res.size() == height) {
            res.add(new ArrayList<Integer>());
        }
        res.get(height).add(x.val);
        if (x.children != null) {
            for (Node child : x.children) {
                levelOrder(child, res, height + 1);
            }
        }
    }
}

File: 1/2020.09.22/429.N-ary Tree Level Order Traversal/python/solution.py
class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        q, res = [root], []
        while any(q):
            res.append([node.val for node in q])
            q = [child for node in q for child in node.children if child]
        return res
